match2/commons/starcraft_starcraft2/match_group_util_starcraft.lua

```diff
@@ -104,22 +104,22 @@ function StarcraftMatchGroupUtil.populateOpponents(match)
 	local opponents = match.opponents
 
 	for _, opponent in ipairs(opponents) do
-		opponent.isArchon = Logic.readBool(opponent.extradata.isarchon)
-		opponent.score2 = tonumber(opponent.extradata.score2)
-		opponent.status2 = opponent.extradata.score2 and 'S' or nil
-		opponent.placement2 = tonumber(opponent.extradata.placement2)
+		opponent.isArchon = Logic.readBool(Table.extract(opponent.extradata, 'isarchon'))
+		opponent.score2 = tonumber(Table.extract(opponent.extradata, 'score2'))
+		opponent.status2 = Table.extract(opponent.extradata, 'score2') and 'S' or nil
```

Comment:
in line 108 score2 is kicked from extradata
in line 109 you try to access it again (if it exists) ... it will never exist at that point
suggestion for line 109: `opponent.status2 = (not Logic.isEmpty(opponent.score2)) and 'S' or nil`
alternative: do not extract score2 in line 108, just access it